### ISSUE TITLE
[Bug] Error during parallel function calls.

### ISSUE THREAD

**Issue by jtokash** [TengHu/ActionWeaver#112]
During parallel function calls, I get this error after all the function calls complete.  Below is my test code and the full output.

Versions
- actionweaver      0.0.20
- openai            1.5.0

Error:
```
File "[path to dev folder]/TestingActionWeaver/.venv/lib/python3.9/site-packages/actionweaver/llms/openai/tools/chat.py", line 319, in new_create
    tools, (stop, resp) = OpenAIChatCompletion._invoke_tool(
ValueError: too many values to unpack (expected 2)

```

Test code:
```
from actionweaver import action
from actionweaver.llms.openai.tools.tokens import TokenUsageTracker
from actionweaver.llms import patch
from openai import OpenAI
import json

@action(name="GetStationZipcode")
def get_station_zipcode(station: str) -> str:
    """
    Use this for getting the zipcode of a given station.
    
    :return: The zipcode for the given station.
    """
    print("> in get_station_zipcode")

    return f"The zipcode for {station} is 8675309"

def ask(question, stream=False):
    print(f"stream={stream} Asking: {question}")
    messages = [
        {"role": "system", "content": """
            You are an expert on a specific set of stations
        
        """},
        {"role": "user", "content": "The stations known to you are: AAA, BBB, CCC, DDD, EEE"},
        {"role": "user", "content": question}
    ]
    openai_client = patch(OpenAI())
    response = openai_client.chat.completions.create(
        model="gpt-4-1106-preview",
        messages=messages,
        actions = [get_station_zipcode],
        token_usage_tracker = TokenUsageTracker(5000),
        stream=stream,
        max_tokens=2800
    )
    print(messages)
    print()
    if stream:
        for chunk in response:
            if chunk.choices[0].delta.content:
                print(chunk.choices[0].delta.content, end="", flush=True)
    else:
        print(response.choices[0].message.content)
    print(f"\n\nFinished Asking: {question}")

ask("What is the zipcode for station AAA?")
ask("What is the zipcode for station AAA?", stream=True)

ask("What are the zipcodes for all stations you know of?")
# Traceback (most recent call last):
#   File "[path to dev folder]/TestingActionWeaver/main.py", line 54, in <module>
#     ask("What are the zipcodes for all stations you know of?")
#   File "[path to dev folder]/TestingActionWeaver/main.py", line 33, in ask
#     response = openai_client.chat.completions.create(
#   File "[path to dev folder]/TestingActionWeaver/.venv/lib/python3.9/site-packages/actionweaver/llms/openai/tools/chat.py", line 319, in new_create
#     tools, (stop, resp) = OpenAIChatCompletion._invoke_tool(
# ValueError: too many values to unpack (expected 2)

ask("What are the zipcodes for all stations you know of?", stream=True)

```

Full output:
```
python main.py
stream=False Asking: What is the zipcode for station AAA?
> in get_station_zipcode
[{'role': 'system', 'content': '\n            You are an expert on a specific set of stations\n        \n        '}, {'role': 'user', 'content': 'The stations known to you are: AAA, BBB, CCC, DDD, EEE'}, {'role': 'user', 'content': 'What is the zipcode for station AAA?'}, ChatCompletionMessage(content=None, role='assistant', function_call=None, tool_calls=[ChatCompletionMessageToolCall(id='call_PjP28Kfwt0xJbjAlu8dNt8Pk', function=Function(arguments='{"station":"AAA"}', name='GetStationZipcode'), type='function')]), {'tool_call_id': 'call_PjP28Kfwt0xJbjAlu8dNt8Pk', 'role': 'tool', 'name': 'GetStationZipcode', 'content': 'The zipcode for AAA is 8675309'}]

The zipcode for station AAA is 8675309.


Finished Asking: What is the zipcode for station AAA?
stream=True Asking: What is the zipcode for station AAA?
> in get_station_zipcode
[{'role': 'system', 'content': '\n            You are an expert on a specific set of stations\n        \n        '}, {'role': 'user', 'content': 'The stations known to you are: AAA, BBB, CCC, DDD, EEE'}, {'role': 'user', 'content': 'What is the zipcode for station AAA?'}, ChatCompletionMessage(content=None, role='assistant', function_call=None, tool_calls=[ChatCompletionMessageToolCall(id='call_QcPYMpTvInw9RTfokBOq09H8', function=Function(arguments='{"station":"AAA"}', name='GetStationZipcode'), type='function', index=0)]), {'tool_call_id': 'call_QcPYMpTvInw9RTfokBOq09H8', 'role': 'tool', 'name': 'GetStationZipcode', 'content': 'The zipcode for AAA is 8675309'}]

The zipcode for station AAA is 8675309.

Finished Asking: What is the zipcode for station AAA?
stream=False Asking: What are the zipcodes for all stations you know of?
> in get_station_zipcode
> in get_station_zipcode
> in get_station_zipcode
> in get_station_zipcode
> in get_station_zipcode
Traceback (most recent call last):
  File "[path to dev folder]/TestingActionWeaver/main.py", line 55, in <module>
    ask("What are the zipcodes for all stations you know of?")
  File "[path to dev folder]/TestingActionWeaver/main.py", line 34, in ask
    response = openai_client.chat.completions.create(
  File "[path to dev folder]/TestingActionWeaver/.venv/lib/python3.9/site-packages/actionweaver/llms/openai/tools/chat.py", line 319, in new_create
    tools, (stop, resp) = OpenAIChatCompletion._invoke_tool(
ValueError: too many values to unpack (expected 2)
```

**Comment by TengHu** [comment]
Thank you, looking into it.

**Comment by TengHu** [comment]
Just released 0.0.21, the bug should be fixed.  Also, you can create get_station_zipcodes instead of have llm invoking multiple functions. Feel free to reopen if you run into any new issue.

```python
from actionweaver import action
from actionweaver.utils.tokens import TokenUsageTracker
from actionweaver.llms import patch
from openai import OpenAI
import json
from typing import List

@action(name="GetStationZipcode")
def get_station_zipcodes(stations: List[str]) -> str:
    """
    Retrieve the zipcodes for a list of stations.

    :param stations: A list of station names for which to retrieve zipcodes.
    :type stations: List[str]
    """

    print("> in get_station_zipcodes", stations)
        
    def get_station_zipcode(station: str) -> str:
        print("> in get_station_zipcode")
        return f"The zipcode for {station} is 8675309"
    
    return [get_station_zipcode(station) for station in stations]


def ask(question, stream=False):
    print(f"stream={stream} Asking: {question}")
    messages = [
        {"role": "system", "content": """
            You are an expert on a specific set of stations
        
        """},
        {"role": "user", "content": "The stations known to you are: AAA, BBB, CCC, DDD, EEE"},
        {"role": "user", "content": question}
    ]
    openai_client = patch(OpenAI())
    response = openai_client.chat.completions.create(
        model="gpt-4-1106-preview",
        messages=messages,
        actions = [get_station_zipcodes],
        token_usage_tracker = TokenUsageTracker(5000),
        stream=stream,
        max_tokens=2800
    )
    print(messages)
    print()
    if stream:
        for chunk in response:
            if chunk.choices[0].delta.content:
                print(chunk.choices[0].delta.content, end="", flush=True)
    else:
        print(response.choices[0].message.content)
    print(f"\n\nFinished Asking: {question}")

ask("What is the zipcode for station AAA?")
ask("What is the zipcode for station AAA?", stream=True)
ask("What are the zipcodes for all stations you know of?", stream=True)

```

**Comment by jtokash** [comment]
Perfect, thanks!
